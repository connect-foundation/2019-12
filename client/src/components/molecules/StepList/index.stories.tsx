import React from 'react';

import StepList from '.';
import { array, number } from '@storybook/addon-knobs';

export default {
  title: 'Molecules / StepList',
};

export const index: React.FC = () => {
  return (
    <StepList
      steps={array('steps', ['a', 'b', 'c'])}
      pivot={number('pivot', 1)}
    />
  );
};
