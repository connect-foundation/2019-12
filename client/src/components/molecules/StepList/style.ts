import styled, { css } from 'styled-components';
import { theme, ifProp, palette } from 'styled-tools';

const StepText = css`
  ${theme('fontStyle.subtitle2')};
  font-weight: bold;
  color: ${ifProp(
    'highlight',
    palette('grayscale', 1),
    palette('grayscale', 3),
  )};
`;

interface StepProps {
  highlight?: boolean;
}

export const Step = styled.span<StepProps>`
  ${StepText};
`;

export const StepArrow = styled.span<StepProps>`
  ${StepText};
  padding: 0rem 2rem;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;
