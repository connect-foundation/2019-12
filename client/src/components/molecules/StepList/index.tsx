import React from 'react';

import * as S from './style';

interface Props {
  /** steps string array */
  steps: string[];
  /** pivot step */
  pivot: number;
}

const makeSteps = (steps: string[], pivot: number) => {
  return steps.map((step, index, arr) => (
    <>
      <S.Step highlight={index <= pivot}>{step}</S.Step>
      {index < arr.length - 1 && (
        <S.StepArrow highlight={index + 1 <= pivot}>&lt;</S.StepArrow>
      )}
    </>
  ));
};

function StepList({ steps, pivot }: Props): React.ReactElement {
  return <S.Container>{makeSteps(steps, pivot - 1)}</S.Container>;
}

export default StepList;
