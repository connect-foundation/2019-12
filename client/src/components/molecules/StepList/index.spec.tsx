import React from 'react';
import { mount } from 'enzyme';
import StepList from '.';

describe('Molecules / StepList', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<StepList steps={['a', 'b', 'c']} pivot={1} />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
