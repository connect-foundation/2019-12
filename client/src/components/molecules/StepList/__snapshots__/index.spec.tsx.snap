// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules / StepList [SNAPSHOT] 렌더링 1`] = `
.c1 {
  font-weight: bold;
}

.c2 {
  font-weight: bold;
  padding: 0rem 3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<StepList
  pivot={1}
  steps={
    Array [
      "a",
      "b",
      "c",
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.span
          highlight={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    ";
  font-weight: bold;
  color: ",
                    [Function],
                    ";
",
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            highlight={true}
          >
            <span
              className="c1"
            >
              a
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          highlight={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    ";
  font-weight: bold;
  color: ",
                    [Function],
                    ";
",
                    ";
  padding: 0rem 3rem;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            highlight={false}
          >
            <span
              className="c2"
            >
              &lt;
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          highlight={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    ";
  font-weight: bold;
  color: ",
                    [Function],
                    ";
",
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            highlight={false}
          >
            <span
              className="c1"
            >
              b
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          highlight={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    ";
  font-weight: bold;
  color: ",
                    [Function],
                    ";
",
                    ";
  padding: 0rem 3rem;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            highlight={false}
          >
            <span
              className="c2"
            >
              &lt;
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          highlight={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    ";
  font-weight: bold;
  color: ",
                    [Function],
                    ";
",
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            highlight={false}
          >
            <span
              className="c1"
            >
              c
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</StepList>
`;
